lib/polyfill/mediasource.js

```diff
@@ -164,6 +164,15 @@ shaka.polyfill.MediaSource = class {
       const actualContainer = shaka.util.MimeUtils.getContainerType(mimeType);
       return actualContainer != container && isTypeSupported(mimeType);
     };
+
+    if (window.ManagedMediaSource) {
+      const isTypeSupportedManaged = window.ManagedMediaSource.isTypeSupported;
```

Comment:
This won't necessarily work on every web implementation.  It may happen to work now, but unbinding the method from the class will cause issues if the implementation uses "this" internally.  Even static methods can have this issue, since "this" has a meaning in static methods since ES6.  Whether or not "this" is used, in JS or C++, is an implementation detail we have no visibility into that could change at any time.

The same is true on const `isTypeSupported = MediaSource.isTypeSupported` above.

Reply:
And what solution do you propose?

Reply:
Either of these would be safe:

```js
// Bind it early
const isTypeSupportedManaged =
    ManagedMediaSource.isTypeSupported.bind(ManagedMediaSource);
```


```js
  // Or provide a "this" context at the call site.
  return actualContainer != container &&
      isTypeSupportedManaged.call(ManagedMediaSource, mimeType);
```

Reply:
Done!